Assets/MRTK/Providers/LeapMotion/Editor/ConfigurationChecker/LeapMotionConfigurationChecker.cs

```diff
@@ -0,0 +1,263 @@
+                Dictionary<string, string[]> leapEditorDirectories = new Dictionary<string, string[]>
+                {
+                    { "LeapMotion.Core.Editor", new string[] { "LeapMotion" } },
+                    { "LeapMotion.Core.Scripts.Animation.Editor", new string[] { "LeapMotion", "LeapMotion.Core.Editor", "LeapMotion.Core.Scripts.Utils.Editor" } },
+                    { "LeapMotion.Core.Scripts.Attachments.Editor", new string[] { "LeapMotion", "LeapMotion.Core.Editor" } },
+                    { "LeapMotion.Core.Scripts.Attributes.Editor", new string[] { "LeapMotion" } },
+                    { "LeapMotion.Core.Scripts.DataStructures.Editor", new string[] { "LeapMotion" } },
+                    { "LeapMotion.Core.Scripts.EditorTools.Editor", new string[] { "LeapMotion", "LeapMotion.Core.Scripts.Utils.Editor" } },
+                    { "LeapMotion.Core.Scripts.Utils.Editor", new string[] { "LeapMotion", "LeapMotion.Core.Editor" } },
+                    { "LeapMotion.Core.Scripts.XR.Editor", new string[] { "LeapMotion", "LeapMotion.Core.Editor" } }
+                };
+
+                string pathDifference = GetPathDifference();
+
+                foreach (KeyValuePair<string, string[]> leapAsmDef in leapEditorDirectories)
+                {
+                    // Convert asmdef name to a path
+                    string leapAsmDefPath = leapAsmDef.Key.Replace('.', '/');
+
+                    string leapAsmDefFilename = string.Concat(leapAsmDef.Key, ".asmdef");
+
+                    string fullLeapAsmDefPath = Path.Combine(Application.dataPath, pathDifference, leapAsmDefPath, leapAsmDefFilename);
+
+                    if (!File.Exists(fullLeapAsmDefPath))
+                    {
+                        // Create and save the new asmdef
+                        AssemblyDefinition leapEditorAsmDef = new AssemblyDefinition
+                        {
+                            Name = leapAsmDef.Key,
+                            References = leapAsmDef.Value,
+                            IncludePlatforms = new string[] { "Editor" }
+                        };
+
+                        leapEditorAsmDef.Save(fullLeapAsmDefPath);
+                    }
+                }
+
+                AssetDatabase.Refresh();
+            }
+        }
+
+        [MenuItem("Mixed Reality Toolkit/Utilities/Update/Configure CSC File for Leap Motion", false, 0)]
+        static void UpdateCSC()
+        {
+            string fileName = Path.Combine(Application.dataPath, "csc.rsp");
+
+            FileInfo file = new FileInfo(fileName);
+
+            bool readOnly = (file.Exists) ? file.IsReadOnly : false;
+            if (readOnly)
+            {
+                file.IsReadOnly = false;
+            }
+
+            Debug.Log($"Saving {fileName}");
+            using (StreamWriter writer = new StreamWriter(fileName, false))
+            {
+                writer.WriteLine("-warnaserror+\n-nowarn:1701,1702,618,649");
```

Comment:
One thing to be careful about here is that, this code assumes that the file only contains:

-warnaserror+\n-nowarn:1701,1702

What this means is if we ever change csc.rsp, this line of code will overwrite things to a now subset of the new stuff (plus 618 and 649).

This is a general problem around mutation where you're trying to do an additive (single time, repeatable but safe to repeat because it doesn't mess things up), and for that assuming exactly what's already there is dangerous.

There are a few options:

1. Update the code to parse out the nowarn part, and just write out that section as an additive (i.e. figure out the numbers, then add 618 and 649 to the array, and then write it back out).
2. Verify ahead of time that the nowarn line ONLY contains "-nowarn:1701,1702"

Reply:
I went with a version of option 1. The nowarn warning numbers are stored, 618 and 649 are now additive instead of assuming what the csc file contains.